### PR TITLE
feat(core): Add prod asset versioning

### LINKED ISSUE #1509: Include asset versioning
Since MEAN app depend heavily on Javascript files, it would be better if assets like CSS, JS and image files are versioned if they are changed. It helps it busting cache issues. In ASP.NET bundling provide minification and versioning. I was searching the same for MEAN and reached to [gulp-rev](https://github.com/sindresorhus/gulp-rev) npm module. But I am new to MEAN and I find it hard to integrate. If any one has used then please guide me for it. I also request to integrate it MEANJS itself.

### PR DESCRIPTION
**Pull request by sujeethk**
feat(core): Add prod asset versioning

Asset versioning for prod - idea from #1509

Fixes #1509

### CODE REVIEW

**Comment by sujeethk** [comment]
maxAge is currently set to 1 day for all the static routes assuming vendor assets will not change in a day. 

Do we need asset versioning for vendor assets as well? If so, they would have to be moved to dist folder and versioned. If this is accomplished then maxAge can be set to a year or more.

File: gulpfile.js [hunk 1]
```diff
@@ -24,6 +24,8 @@ var _ = require('lodash'),
   protractor = require('gulp-protractor').protractor,
   webdriver_update = require('gulp-protractor').webdriver_update,
   webdriver_standalone = require('gulp-protractor').webdriver_standalone,
+  rev = require('gulp-rev'),
```
Comment (rhutchison) [inline gulpfile.js]: why not use plugins.rev() below?

Comment (sujeethk) [inline gulpfile.js]: Valid point :)

**Comment by lirantal** [comment]
Looks good, thanks.

**Pull request merged**